### DISCUSSION TITLE
The document page is too slow.

### DISCUSSION THREAD

**Discussion by usuall** [zauberzeug/nicegui#1720] [General]
The document page is too slow.
Chrome browser freezes.
I think it's slowing down because I'm even showing examples on one page. Please improve it.
(My environment is Windows 11, Cpu: Ryzen 5xxxx, Mem:40G.)

**Comment by rodja** [comment]
Sorry you experience such issues. Is that always happening? Does it happen when simply opening the page once? We can not reproduce these freezes.

**Reply by usuall** [reply]
It always happens when you open the document page. How about separating the document?
As the sample code loads, it slows down a lot, and the chrome freezes and OS resource ... stops.

**Comment by rodja** [comment]
Whats your region? We have multiple servers around the world hosting the website. Maybe your region is not sufficiently covered? Or do other users also experience this issue?

**Comment by Declow** [comment]
I currently have the same issue.
I live in Denmark and sometimes the whole page just reloads for no reason. Currently using the latest version of firefox.

To extend a bit more on this. This is from a Mac M1 with a 500mb wifi connection.
<img width="828" alt="Screenshot 2023-10-12 at 12 43 34" src="https://github.com/zauberzeug/nicegui/assets/4069574/0f9d593b-e7c4-4550-844a-c85aec9ab97d">

**Reply by stdusr** [reply]
Second that, sometimes it takes a few minutes for the documentation to load

**Comment by falkoschindler** [comment]
Thanks, everyone, for your feedback!
You are absolutely right, the documentation page is too slow. We see multiple potential reasons:

- We're experiencing growing visitor counts. Although we try to keep up with the demand by adding more and more servers, it might compromise the performance while we try to find the right configuration.
- In beginning of October we had to migrate to Fly.io V2, which caused quite some trouble and some downtime. It is still not running as smoothly as we wish.
- Regarding the occasional reconnects, we're working on an improvement in PR https://github.com/zauberzeug/nicegui/pull/1762.
- Last but not least, there is (probably) a memory leak that causes an increase in memory consumption on our servers that eventually leads to a restart. We are investigating this issue on https://github.com/zauberzeug/nicegui/issues/1771 and would appreciate any help.

**Reply by rodja** [reply]
To bridge the time until nicegui.io is fast again, you can fire up a local copy:

```bash
docker run --rm -it -p 8080:8080 nicegui:latest
```

**Comment by rodja** [comment]
Ah I just found another issue with our fly.io hosting: We configured it so that it automatically stops unused machines and starts them if traffic increases as described in https://fly.io/docs/apps/autostart-stop/. But I now discovered that machines which are stopped due to downscaling trigger a health-check alert and are started again. Then the auto-stop will stop it again, which in turn triggers a health-check alert. On and on it goes. Maybe the long page loading issues arise from such machines which constantly stop and restart...

I've contacted the fly.io support and hope to have an answer in the next hours.

**Reply by rodja** [reply]
I've temporarily modified our configuration to have a fixed number of machines (i.e. scale count = min_machines_running) and disabled auto-stopping. Hope this helps until we can go back to auto stop/start.

**Reply by Declow** [reply]
It seems a bit better. The load time can still be quite long. But the browser is waiting for setting up the initial connection. This can still take some time. Unsure why that is.
But the response from the server is a lot quicker. I'm getting timings for the document from 1 sec to 4 seconds. The examples now load way more consistently and faster.

**Reply by frankhuurman** [reply]
![image](https://github.com/zauberzeug/nicegui/assets/19980966/ce1d6a46-25e3-45a1-b119-3015eec4670e)

Seems a bit better indeed, around 4 seconds on average for me with still the occasional reconnect or 'page refresh while it looked like it was done loading' but I'm hoping the before mentioned PR helps with that!

Before it could take between 7-15 seconds so I'll take it haha :+1:

**Comment by rodja** [comment]
I've found another cause: fly.io seems to create unexpected pycache files like `/app/website/more_documentation/__pycache__/carousel_documentation.cpython-311.pyc.140641625254784`. These where not excluded from auto-reload so the creation of pycache files on server start resulted in an direct reload. CRAZY! I just pushed 652f3aca06ab which disables auto-reloading in the production environment. Can't wait to deploy this stuff.

**Reply by falkoschindler** [reply]
Strange. The "uvicorn reload includes" default to *.py, i.e. not including such cache files. 🤔

**Reply by rodja** [reply]
Ah, my error. We do not reload but watchfiles takes a long time to detect the change. And this degrades server performance.

**Comment by rodja** [comment]
Ok. We just released NiceGUI 1.3.17 and deployed the code to https://nicegui.io. Please check it out and leave your feedback. We have improved some parts. But the main issue with scaling up/down is still being investigated by the fly.io technicans.

**Comment by TeeFiX42** [comment]
This morning it's very painfull to use the documentation.
Working from Belgium with high speed connection.
f.e.: https://nicegui.io/documentation/section_styling_appearance
took 17 seconds to load and allmost every example is still spinning.

**Reply by rodja** [reply]
Sorry there was an deployment issue. Should be fixed now.
